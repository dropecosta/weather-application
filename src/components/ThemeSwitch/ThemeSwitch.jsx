import React from 'react';
import './ThemeSwitch.scss';

const ThemeSwitch = ({ isCheck, handleDarkMode, onColor }) => {
  return (
    <>
      <input
        checked={isCheck}
        onChange={handleDarkMode}
        className="switch-checkbox"
        id={`switch-darkmode`}
        type="checkbox"
      />
      <label
        style={{ background: isCheck && onColor }}
        className="switch-label"
        htmlFor={`switch-darkmode`}
      >
        <span className={`switch-button`} />
      </label>
    </>
  );
};

export default ThemeSwitch
