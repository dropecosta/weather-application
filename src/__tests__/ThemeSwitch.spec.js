import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import ThemeSwitch from "../components/ThemeSwitch/ThemeSwitch";

describe("ThemeSwitch component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<ThemeSwitch />);
    expect(container.firstChild).toHaveClass("switch-checkbox");
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<ThemeSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
