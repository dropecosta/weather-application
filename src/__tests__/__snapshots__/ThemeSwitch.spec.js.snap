// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch component tests should match the snapshot 1`] = `
Array [
  <input
    className="switch-checkbox"
    id="switch-darkmode"
    type="checkbox"
  />,
  <label
    className="switch-label"
    htmlFor="switch-darkmode"
    style={
      Object {
        "background": undefined,
      }
    }
  >
    <span
      className="switch-button"
    />
  </label>,
]
`;
